pyomo/core/plugins/transform/scaling.py

```diff
@@ -0,0 +1,269 @@
+        
+        # if the scaling_method is 'user', get the scaling parameters from the suffixes
+        if self._scaling_method == 'user':
+            # perform some checks to make sure we have the necessary suffixes
+            if type(model.component('scaling_factor')) is not Suffix:
+                raise ValueError("ScaleModel transformation called with scaling_method='user'"
+                                 ", but cannot find the suffix 'scaling_factor' on the model")
+
+            # get the scaling factors
+            for c in model.component_data_objects(ctype=(Var,Constraint,Objective), descend_into=True):
+                component_scaling_factor_map[c] = self._get_float_scaling_factor(model, c)
+
+        else:
+            raise ValueError("ScaleModel transformation: unknown scaling_method found"
+                             "-- supported values: 'user' ")
+
+        # scale the variable bounds and values and build the
+        # variable substitution map for scaling vars in constraints
+        variable_substitution_map = ComponentMap()
+        for variable in [var for var in model.component_objects(ctype=Var, descend_into=True)]:
+            # set the bounds/value for the scaled variable
+            for k in variable:
+                v = variable[k]
+                scaling_factor = component_scaling_factor_map[v]
+                variable_substitution_map[v] = v/scaling_factor
+
+                if v.lb is not None:
+                    v.setlb(v.lb * scaling_factor)
+                if v.ub is not None:
+                    v.setub(v.ub * scaling_factor)
+                if scaling_factor < 0:
+                    temp = v.lb
+                    v.setlb(v.ub)
+                    v.setub(temp)
+
+                if v.value is not None:
+                    v.value = value(v) * scaling_factor
+
+        # scale the objectives/constraints and perform the scaled variable substitution
+        scale_constraint_dual = False
+        if type(model.component('dual')) is Suffix:
+            scale_constraint_dual = True
+            
+        for component in model.component_objects(ctype=(Constraint,Objective), descend_into=True):
+            for k in component:
+                c = component[k]
+                # perform the constraint/objective scaling and variable sub
+                scaling_factor = component_scaling_factor_map[c]
+                if isinstance(c, _ConstraintData):
+                    _substitute_var_in_constraint(c, variable_substitution_map)
+
+                    if c._lower is not None:
+                        c._lower = c._lower * scaling_factor
+                    if c._upper is not None:
+                        c._upper = c._upper * scaling_factor
+
+                    if scaling_factor < 0:
+                        # swap inequalities
+                        temp = c._lower
+                        c._lower = c._upper
+                        c._upper = temp
```

Comment:
The Pythonic Swap is:
```
c._lower, c._upper = c._upper, c._lower
```
(it is also faster than using the intermediate variable)